src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -1751,42 +1758,44 @@ private void hitBlockWithToolInHand(@Nullable final BlockPos blockPos, final boo
 
         this.swingArm(this.getActiveHand());
 
-        final IBlockState blockState = world.getBlockState(blockPos);
+        final IBlockState blockState = CompatibilityUtils.getWorld(this).getBlockState(blockPos);
         final Block block = blockState.getBlock();
         if (breakBlock)
         {
-            if (!world.isRemote)
+            if (!CompatibilityUtils.getWorld(this).isRemote)
             {
                 MineColonies.getNetwork().sendToAllAround(
-                        new BlockParticleEffectMessage(blockPos, world.getBlockState(blockPos), BlockParticleEffectMessage.BREAK_BLOCK),
-                        new NetworkRegistry.TargetPoint(world.provider.getDimension(), blockPos.getX(), blockPos.getY(), blockPos.getZ(), BLOCK_BREAK_SOUND_RANGE));
+                        new BlockParticleEffectMessage(blockPos, CompatibilityUtils.getWorld(this).getBlockState(blockPos), BlockParticleEffectMessage.BREAK_BLOCK),
+                        new NetworkRegistry.TargetPoint(CompatibilityUtils.getWorld(this).provider.getDimension(),
+                        blockPos.getX(), blockPos.getY(), blockPos.getZ(), BLOCK_BREAK_SOUND_RANGE));
             }
-            world.playSound(null,
+            CompatibilityUtils.getWorld(this).playSound(null,
                     blockPos,
-                    block.getSoundType(blockState, world, blockPos, this).getBreakSound(),
+                    block.getSoundType(blockState, CompatibilityUtils.getWorld(this), blockPos, this).getBreakSound(),
                     SoundCategory.BLOCKS,
-                    block.getSoundType(blockState, world, blockPos, this).getVolume(),
-                    block.getSoundType(blockState, world, blockPos, this).getPitch());
-            world.setBlockToAir(blockPos);
+                    block.getSoundType(blockState, CompatibilityUtils.getWorld(this), blockPos, this).getVolume(),
+                    block.getSoundType(blockState, CompatibilityUtils.getWorld(this), blockPos, this).getPitch());
+            CompatibilityUtils.getWorld(this).setBlockToAir(blockPos);
 
             damageItemInHand(1);
         }
         else
         {
             //todo: might remove this
-            if (!world.isRemote)
+            if (!CompatibilityUtils.getWorld(this).isRemote)
             {
                 MineColonies.getNetwork().sendToAllAround(
                         //todo: correct side
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)